packages/devtools_shared/lib/src/memory/event_sample.dart

```diff
@@ -163,3 +165,35 @@ class EventSample {
       'isEventSnapshotAuto: $isEventSnapshotAuto,'
       'allocationAccumulator: \n   $allocationAccumulator]';
 }
+
+/// Engine's Raster Cache estimates.
+class RasterCache {
+  RasterCache({@required this.layerBytes, @required this.pictureBytes});
+
+  RasterCache.fromJson(Map<String, dynamic> json) {
+    layerBytes = json['layerBytes'];
+    pictureBytes = json['pictureBytes'];
+  }
+
+  static RasterCache empty() => RasterCache(layerBytes: 0, pictureBytes: 0);
+
+  static RasterCache parse(Map<String, dynamic> json) =>
+      json == null ? null : RasterCache.fromJson(json);
+
+  int layerBytes;
+  
+  int pictureBytes;
+
+  Map<String, dynamic> toJson() {
+    final json = <String, dynamic>{};
```

Comment:
why use addAll instead of just returning a map literal directly?

Reply:
Should

  Map<String, dynamic> toJson() => {
        'layerBytes': layerBytes,
        'pictureBytes': pictureBytes,
      };